atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/ValidatingTransactionScopedCache.java

```diff
@@ -27,28 +28,28 @@
 import com.palantir.logsafe.UnsafeArg;
 import java.util.Map;
 import java.util.Objects;
+import java.util.Random;
 import java.util.Set;
-import java.util.concurrent.ThreadLocalRandom;
 import java.util.function.BiFunction;
-import java.util.function.Supplier;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 final class ValidatingTransactionScopedCache implements TransactionScopedCache {
     private static final Logger log = LoggerFactory.getLogger(ValidatingTransactionScopedCache.class);
 
     private final TransactionScopedCache delegate;
-    private final Supplier<Boolean> shouldValidate;
+    private double validationProbability;
+    private final Random random;
 
     @VisibleForTesting
-    ValidatingTransactionScopedCache(TransactionScopedCache delegate, Supplier<Boolean> shouldValidate) {
+    ValidatingTransactionScopedCache(TransactionScopedCache delegate, double validationProbability) {
```

Comment:
Since I made it possible to specify the probability instead of making it static here, no need for the janky supplier.